[CS2103T-W11-3] EatMe

dvrylc opened the pull request:
@jakq
@JunHongT
@walnads
@ROHITREDDYBALAM

dvrylc committed:
Change Travis link and badge in readme

dvrylc committed:
Update Coveralls link and badge in readme

dvrylc committed:
Remove codacy badge from readme

dvrylc committed:
Update documentation to match fork

JunHongT committed:
Update README

wpuffs committed:
Update UserGuide.adoc and Ui.png (#9)

dvrylc committed:
Update About and Contact page (#11)

JunHongT committed:
Fix image conflict

JunHongT committed:
Fix image conflict

dvrylc committed:
Merge remote-tracking branch 'upstream/master'

dvrylc committed:
Fix image extensions

wpuffs committed:
Merge pull request #12 from dvrylc/fix-about-image-extensions

Fix About page image extensions

JunHongT committed:
Merge branch 'edit-README' of https://github.com/JunHongT/main into edit-README

ROHITREDDYBALAM committed:
Merge pull request #10 from JunHongT/edit-README

Update README

jakq committed:
Update Developer Guide (#14)

* Update Developer Guide

Updated Appendix A, B, C, D, E, F

* Update Developer Guide

Edited Appendix  C (remove entry from todo).

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -12,7 +12,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level3/tree/master
+:repoURL: https://github.com/AY1920S1-CS2103T-W11-3/main/tree/master
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Are you sure you are from Team SE-EDU? It will be cool if you are.

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,37 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |person with many food allergies |keep track of dishes I have ordered before |avoid ordering hazardous food
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |foodie |remember places |share this experience with my family and friends
```
This may be a nice to have, but good job anyways

wallacelim commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,37 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |foodie |remember places |share this experience with my family and friends
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *` |digital nomad |tag places with free power + wifi |sit there for extended periods of time
```
Benefit is not immediately obvious, consider rephrasing? :)

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +298,37 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+3.  User requests to mark an eatery as visited in the list
+4.  EatMe shifts the data over
+5.  EatMe toggles to the main mode
+6.  EatMe requests user for whatever information left that is required before saving in the main mode
```
great use case, clear and concise.

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,21 +339,78 @@ Use case ends.
++
+Use case ends.
+
+*Extensions*
```
You may want to format the nesting of your **2a. The List is empty**, **Use case ends.**.

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,21 +339,78 @@ Use case ends.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 eateries without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Main functionalities (CRUD) should work without WiFi.
```
Useful NFR's, good job

Berttwm commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,25 +420,28 @@ _{More to be added}_
+Any place that serves food open to the public (ie, restaurants, cafes, roadside, pushcart)
+
+[[category]] Category::
+Broadly containts cuisines (self-explanatory) and eating place (ie, a buffet, fusion)
```
May want to define what "todo" means as developers may need to understand such terms :)

Berttwm reviewed (commented):
Overall good job. Can consider formatting your use cases using proper ASCII doc syntax for better readability.

ROHITREDDYBALAM committed:
remove email field (#57)

* remove email field

* Remove email field

ROHITREDDYBALAM committed:
Remove phone and email fields (#59)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

jakq committed:
Refactor Person to Eatery (#61) 

* Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

* Removed 2 test cases from EditCommandTest

* Update BetterModelClassDiagram.puml

* Remove Email and Phone class

* Remove Email Phone multiplicity

* Refactor Person to Eatery edit1

*Close issue #55

wpuffs committed:
Add Category class (#62)

Closes #53 🚀 

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Fix coding style violations

* Create test cases (#1)

* Fix coding style violations (#1)

* Fix coding style violations (#2)

* Fix getId() method to return object's id instead of static variable id

* Fix test case that checks category id

* Add Ui.png ~_~

dvrylc committed:
Fix category location

wpuffs committed:
Merge pull request #67 from dvrylc/fix-category-location

Fix category location

dvrylc committed:
Implement 3-column UI (#63)

* Implement 3-column UI, closes #48

- Shift content block to center column
- Add placeholder right column for news feed
- Refactor basic styles

* Update views for Eatery refactor

* Add category placeholder label

dvrylc committed:
Implement basic news feed (#73)

* Implement 3-column UI, closes #48

- Shift content block to center column
- Add placeholder right column for news feed
- Refactor basic styles

* Update views for Eatery refactor

* Add category placeholder label

* Implement basic news feed

Includes:
- Basic news feed functionality
- Tests for new models

* Improve code style

- Fix indentation
- Improve variable names
- Remove unnecessary `if` conditions

wpuffs committed:
Add Tag class, refactor Add and Edit command (#74)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add Tag class

* Add tests for Tag class

* Refactor existing classes

* Refactor existing test classes

* Remove trailing whitespaces

* Fix code to refer to Tag class and not Category class

* Fix ParserUtil class and test class

* Change toString() method of AddressBook

JunHongT committed:
Add mode command (#68)

* Add Mode Command

* Fix checkstyle

* Fix minor bug in JsonSerializableAddressBook.java

* Enable delete command for to-do list

* Fix checkstyle

* Fix minor bug in DeleteCommand

* Create ModeCommandTest

* Fix Checkstyle

* Improve code quality

* Fix test cases

* Improve code quality

jakq committed:
Add CloseCommand (#76)

* Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

* Removed 2 test cases from EditCommandTest

* Update BetterModelClassDiagram.puml

* Remove Email and Phone class

* Remove Email Phone multiplicity

* Refactor Person to Eatery edit1

* Add Close Command

* Restore whiteline to AddressBookTest

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/Logic.java
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/eatery/Eatery.java
#	src/main/java/seedu/address/model/eatery/UniqueEateryList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/JsonAdaptedEatery.java
#	src/main/java/seedu/address/storage/JsonSerializableAddressBook.java
#	src/main/java/seedu/address/ui/EateryCard.java
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/EateryListCard.fxml
#	src/main/resources/view/EateryListPanel.fxml
#	src/main/resources/view/MainWindow.fxml
#	src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/invalidEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/typicalEateriesAddressBook.json
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/address/logic/commands/EditCommandTest.java
#	src/test/java/seedu/address/logic/commands/FindCommandTest.java
#	src/test/java/seedu/address/logic/commands/ListCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
#	src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/model/eatery/EateryTest.java
#	src/test/java/seedu/address/model/eatery/UniqueEateryListTest.java
#	src/test/java/seedu/address/storage/JsonAdaptedEateryTest.java
#	src/test/java/seedu/address/testutil/EateryBuilder.java
#	src/test/java/seedu/address/testutil/EateryUtil.java
#	src/test/java/seedu/address/testutil/EditEateryDescriptorBuilder.java
#	src/test/java/seedu/address/testutil/TypicalEateries.java

* Fix CloseCommand

1. Remove buildClose() from EateryBuilder
2. Add EOF line to EateryListPanel.fxml
3. Fix duplicateEateryAddressBook.json
4. Require nonnull Category for Eatery

* Remove tags from requireAllNonNull in Eatery

dvrylc committed:
Implement threading for post fetching (#78)

* Implement 3-column UI, closes #48

- Shift content block to center column
- Add placeholder right column for news feed
- Refactor basic styles

* Update views for Eatery refactor

* Add category placeholder label

* Implement basic news feed

Includes:
- Basic news feed functionality
- Tests for new models

* Improve code style

- Fix indentation
- Improve variable names
- Remove unnecessary `if` conditions

* Move post-fetching to a separate thread

* Revert unnecessary code shift

jakq committed:
Rework CloseCommand class (#79)

* Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

* Removed 2 test cases from EditCommandTest

* Update BetterModelClassDiagram.puml

* Remove Email and Phone class

* Remove Email Phone multiplicity

* Refactor Person to Eatery edit1

* Add Close Command

* Restore whiteline to AddressBookTest

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/Logic.java
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/eatery/Eatery.java
#	src/main/java/seedu/address/model/eatery/UniqueEateryList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/JsonAdaptedEatery.java
#	src/main/java/seedu/address/storage/JsonSerializableAddressBook.java
#	src/main/java/seedu/address/ui/EateryCard.java
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/EateryListCard.fxml
#	src/main/resources/view/EateryListPanel.fxml
#	src/main/resources/view/MainWindow.fxml
#	src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/invalidEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/typicalEateriesAddressBook.json
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/address/logic/commands/EditCommandTest.java
#	src/test/java/seedu/address/logic/commands/FindCommandTest.java
#	src/test/java/seedu/address/logic/commands/ListCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
#	src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/model/eatery/EateryTest.java
#	src/test/java/seedu/address/model/eatery/UniqueEateryListTest.java
#	src/test/java/seedu/address/storage/JsonAdaptedEateryTest.java
#	src/test/java/seedu/address/testutil/EateryBuilder.java
#	src/test/java/seedu/address/testutil/EateryUtil.java
#	src/test/java/seedu/address/testutil/EditEateryDescriptorBuilder.java
#	src/test/java/seedu/address/testutil/TypicalEateries.java

* Fix CloseCommand

1. Remove buildClose() from EateryBuilder
2. Add EOF line to EateryListPanel.fxml
3. Fix duplicateEateryAddressBook.json
4. Require nonnull Category for Eatery

* Remove tags from requireAllNonNull in Eatery

* Rework execute method of CloseCommand

- Removed the hasExactEatery method and its calls.

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/commands/CloseCommand.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/CloseCommandTest.java

jakq committed:
Update DeveloperGuide with close feature (#82)

* Update Developer Guide

Updated Appendix A, B, C, D, E, F

* Update Developer Guide

Edited Appendix  C (remove entry from todo).

* Update DeveloperGuide.adoc with Close feature

* Remove UndoSequenceDiagram

* Remove trailing whitespaces from DeveloperGuide

* Update DeveloperGuide.adoc

ROHITREDDYBALAM committed:
Review class (#77)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add basic version of Review class

* Resolve merge conflicts with upstream

* Fix checkstyle errors

* Fix failing test cases

* Fix failing tests

* Verify test cases checking for similar eateries

* Make reviews Json readable

* Fix checkstyle errors and json adapted review errors

* Make review changes permanent

* Fix checkstyle errors

* Fix checkstyle errors

* Create set reviews method for eatery

* Catch exceptions while parsing review command

* Override equals method in review class

* Improve code quality

* Remove unused imports

dvrylc committed:
Add `addfeed` command

dvrylc committed:
Add `deletefeed` command

dvrylc committed:
Merge remote-tracking branch 'upstream/master' into feed-commands

dvrylc committed:
Add missing Javadocs

dvrylc committed:
Add tests for feed-related commands

JunHongT committed:
Merge pull request #88 from dvrylc/feed-commands

Add feed-related commands

wpuffs committed:
Update developer guide with add feature (#87)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add activity and sequence diagram for Add command

* Change sequence diagram

* Change activity diagram

* Resize activity diagram

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

JunHongT committed:
Update developer guide with Mode command details (#84)

* Edit Developer Guide for Mode Command

* Update developer guide

ROHITREDDYBALAM committed:
Update developer guide to show review command details (#81)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add details of review command to Developer Guide

* Dg  (#5)

* Rework CloseCommand class (#79)

* Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

* Removed 2 test cases from EditCommandTest

* Update BetterModelClassDiagram.puml

* Remove Email and Phone class

* Remove Email Phone multiplicity

* Refactor Person to Eatery edit1

* Add Close Command

* Restore whiteline to AddressBookTest

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/Logic.java
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/eatery/Eatery.java
#	src/main/java/seedu/address/model/eatery/UniqueEateryList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/JsonAdaptedEatery.java
#	src/main/java/seedu/address/storage/JsonSerializableAddressBook.java
#	src/main/java/seedu/address/ui/EateryCard.java
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/EateryListCard.fxml
#	src/main/resources/view/EateryListPanel.fxml
#	src/main/resources/view/MainWindow.fxml
#	src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/invalidEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/typicalEateriesAddressBook.json
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/address/logic/commands/EditCommandTest.java
#	src/test/java/seedu/address/logic/commands/FindCommandTest.java
#	src/test/java/seedu/address/logic/commands/ListCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
#	src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/model/eatery/EateryTest.java
#	src/test/java/seedu/address/model/eatery/UniqueEateryListTest.java
#	src/test/java/seedu/address/storage/JsonAdaptedEateryTest.java
#	src/test/java/seedu/address/testutil/EateryBuilder.java
#	src/test/java/seedu/address/testutil/EateryUtil.java
#	src/test/java/seedu/address/testutil/EditEateryDescriptorBuilder.java
#	src/test/java/seedu/address/testutil/TypicalEateries.java

* Fix CloseCommand

1. Remove buildClose() from EateryBuilder
2. Add EOF line to EateryListPanel.fxml
3. Fix duplicateEateryAddressBook.json
4. Require nonnull Category for Eatery

* Remove tags from requireAllNonNull in Eatery

* Rework execute method of CloseCommand

- Removed the hasExactEatery method and its calls.

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/commands/CloseCommand.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/CloseCommandTest.java

* Update DeveloperGuide with close feature (#82)

* Update Developer Guide

Updated Appendix A, B, C, D, E, F

* Update Developer Guide

Edited Appendix  C (remove entry from todo).

* Update DeveloperGuide.adoc with Close feature

* Remove UndoSequenceDiagram

* Remove trailing whitespaces from DeveloperGuide

* Update DeveloperGuide.adoc

* Review class (#77)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add basic version of Review class

* Resolve merge conflicts with upstream

* Fix checkstyle errors

* Fix failing test cases

* Fix failing tests

* Verify test cases checking for similar eateries

* Make reviews Json readable

* Fix checkstyle errors and json adapted review errors

* Make review changes permanent

* Fix checkstyle errors

* Fix checkstyle errors

* Create set reviews method for eatery

* Catch exceptions while parsing review command

* Override equals method in review class

* Improve code quality

* Remove unused imports

* Add `addfeed` command

* Add `deletefeed` command

* Add missing Javadocs

* Add tests for feed-related commands

* Update developer guide with add feature (#87)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add activity and sequence diagram for Add command

* Change sequence diagram

* Change activity diagram

* Resize activity diagram

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update developer guide with Mode command details (#84)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update dg

* Update Developer guide

* Make changes to dg

jakq committed:
Make DeveloperGuide more uniform (#92)

* Make DeveloperGuide more uniform

* Fix stylistic errors in DeveloperGuide

dvrylc committed:
Update User Guide (#91)

* Update User Guide

* Improve User Guide

ROHITREDDYBALAM committed:
Add date and prevent users from reviewing closed eateries. (#90)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Include date in reviews

* Prevent users from reviewing closed eateries

* Store date as date object

* Fix checkstyle errors

* Improve code quality

JunHongT committed:
Implement AddTodoCommandParser (#89)

* Implement AddTodoCommandParser for todo list

* Refractor AddressBookParser

* Modify command message for AddTodoCommandParser

* Fix checkstyle

* Create AddTodoCommandParserTest

* Fix checkstyle

* Update AddTodoCommandParserTest

JunHongT committed:
Implement SaveTodoCommand (#93)

* Implement AddTodoCommandParser for todo list

* Refractor AddressBookParser

* Modify command message for AddTodoCommandParser

* Fix checkstyle

* Create AddTodoCommandParserTest

* Fix checkstyle

* Implement SaveTodoCommand

* Implement test cases for SaveTodoCommand and SaveTodoCommandParser

* Fix checkstyle

* Fix checkstyle

* Update SaveTodoCommand and SaveTodoCommandParser

* Update SaveTodoCommandTest

dvrylc committed:
Implement feed caching (#95)

* Implement feed caching

* Refactor feed cache implementation

* Improve cache data structure

* Remove unnecessary debug statement

jakq committed:
Add ReopenCommand (#96)

* Refactor Person to Eatery

v1.0 with 3 EditCommandTest failures.

* Removed 2 test cases from EditCommandTest

* Update BetterModelClassDiagram.puml

* Remove Email and Phone class

* Remove Email Phone multiplicity

* Refactor Person to Eatery edit1

* Add Close Command

* Restore whiteline to AddressBookTest

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/Logic.java
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/ParserUtil.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/eatery/Eatery.java
#	src/main/java/seedu/address/model/eatery/UniqueEateryList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/JsonAdaptedEatery.java
#	src/main/java/seedu/address/storage/JsonSerializableAddressBook.java
#	src/main/java/seedu/address/ui/EateryCard.java
#	src/main/java/seedu/address/ui/MainWindow.java
#	src/main/resources/view/EateryListCard.fxml
#	src/main/resources/view/EateryListPanel.fxml
#	src/main/resources/view/MainWindow.fxml
#	src/test/data/JsonSerializableAddressBookTest/duplicateEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/invalidEateryAddressBook.json
#	src/test/data/JsonSerializableAddressBookTest/typicalEateriesAddressBook.json
#	src/test/java/seedu/address/logic/LogicManagerTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/address/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/address/logic/commands/EditCommandTest.java
#	src/test/java/seedu/address/logic/commands/FindCommandTest.java
#	src/test/java/seedu/address/logic/commands/ListCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
#	src/test/java/seedu/address/logic/parser/EditCommandParserTest.java
#	src/test/java/seedu/address/logic/parser/ParserUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/model/eatery/EateryTest.java
#	src/test/java/seedu/address/model/eatery/UniqueEateryListTest.java
#	src/test/java/seedu/address/storage/JsonAdaptedEateryTest.java
#	src/test/java/seedu/address/testutil/EateryBuilder.java
#	src/test/java/seedu/address/testutil/EateryUtil.java
#	src/test/java/seedu/address/testutil/EditEateryDescriptorBuilder.java
#	src/test/java/seedu/address/testutil/TypicalEateries.java

* Fix CloseCommand

1. Remove buildClose() from EateryBuilder
2. Add EOF line to EateryListPanel.fxml
3. Fix duplicateEateryAddressBook.json
4. Require nonnull Category for Eatery

* Remove tags from requireAllNonNull in Eatery

* Rework execute method of CloseCommand

- Removed the hasExactEatery method and its calls.

* Merge remote-tracking branch 'upstream/master' into close-open-eatery

# Conflicts:
#	src/main/java/seedu/address/logic/commands/CloseCommand.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/CloseCommandTest.java

* Add ReopenCommand

* Rename variables in ReopenCommand

* Change 0 to getZeroBased

* Fix Close and Reopen command

wpuffs committed:
Fix EditCommand, added more tests for Category (#98)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Fix coding style violations

* Create test cases (#1)

* Fix coding style violations (#1)

* Fix coding style violations (#2)

* Fix getId() method to return object's id instead of static variable id

* Fix test case that checks category id

* Add Ui.png ~_~

* Fix edit command

* Made changes to the message usage of Add command

* Add code to test for category

* Recommit Category class

* Delete duplicate Category class

* Remove duplicate CategoryTest class

JunHongT committed:
Refactor test and sample data (#112)

dvrylc committed:
Implement show command (#100)

* Implement show command

- Add `show` command and tests
- Update UI to accomodate show view
- Implement dark mode and general UI cleanup

* Fix broken tests caused by upstream method rename

* Add missing parser test

* Fix list item overflow

dvrylc committed:
Add missing serializable `FeedPost` (#103)

dvrylc committed:
Update docs (#115)

JunHongT committed:
Update todo commands in UG (#117)

dvrylc committed:
Update docs to indicate commands coming in v1.4 (#118)

wpuffs committed:
Fix issues with add command  (#156)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Edit regex to accept category names with one character

* Edit regex to accept tag names with less than 3 characters and with "-"

* Update UserGuide.adoc

Update the tags part

ROHITREDDYBALAM committed:
Creating add tag and remove tag commands (#155)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* create addtag command

* Add parser for addtag command

* Retain reviews for edited eateries

* Make tag set unmodifiable

* fix bugs related to add tag command

* change ug to remove changeaddress command

* add removetag command

* fix documentation errors

JunHongT committed:
Fix bug for Save Command Ui (#135)

JunHongT committed:
Restrict add review (#157)

* Fix bug for Save Command Ui

* Add restriction for review command

JunHongT committed:
Edit URL in help window (#158)

* Fix bug for Save Command Ui

* Add restriction for review command

* Change url of help window to EatMe UserGuide

JunHongT committed:
Edit user guide (#159)

* Fix bug for Save Command Ui

* Add restriction for review command

* Change url of help window to EatMe UserGuide

* Update user guide

* Update user guide

dvrylc committed:
Add reviews to `show` command

JunHongT committed:
Merge pull request #162 from dvrylc/result-display-reviews

Add reviews to `show` command

dvrylc committed:
Improve feed post UI (#165)

dvrylc committed:
Add visual indicator for closed eateries (#164)

* Add visual indicator for closed eateries

* Improve closed eatery name indicator

- Switch from hightlight and Xs to a simple text indicator

JunHongT committed:
Edit user guide (#167)

* Fix bug for Save Command Ui

* Add restriction for review command

* Change url of help window to EatMe UserGuide

* Update user guide

* Update user guide

* Increase depth of content page

jakq committed:
Add LoadCommand feature (#166)

* Add ChangeCommand to EatMe

* Add JavaDocs to checkFile

* Fix import style.

* Refactor ChangeCommand to LoadCommand

* Fix stylistic errors

* Add LoadCommandTest

* Improve overall `load` command

* Add missing EOF newline

* Fix example error

* Improve `load` command parser tests

dvrylc committed:
Implement attribute-based finding (#168)

* Implement attribute-based finding

* Improve test cases for FindCommand

wpuffs committed:
Create tests for add and remove tag commands (#169)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add activity and sequence diagram for Add command

* Change sequence diagram

* Change activity diagram

* Resize activity diagram

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update photos

* Fix RemoveTagCommandParser; it was not implementing the Parser class

* Add tests for add and remove tag commands

* Fix coding style violations

ROHITREDDYBALAM committed:
Ug changes (#173)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* make changes to ug

* Include examples for edit command in ug

* fix errors in ug

* add images in ug

wpuffs committed:
Implement stats command and add stats window (#111)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Implement stats command i.e. the statistics feature

* Fix line ending

* Change the logic of how stats command works

* Remove existing stats exceptions and add new ones

* Add stylesheets related to the charts and graphs generated by stats command

* Modify and add new custom charts and graphs

* Fix coding style violations

* Minor changes to exception handling

* Create tests for stats command

* Fix coding style violations

* Create tests for parsing stats command

* Fixed unhandled exception when review is less than 3

* Fix coding style violations

* Rename stats exception classes to follow convention

* Make appropriate changes based on code review

* Attempt to fix wrong line ending

* Fix CRLF line endings

* Optimize compareTo for priority queues

* THIRD attempt at fixing line ending issue

JunHongT committed:
Edit SaveTodoCommand in UserGuide (#175)

* Fix bug for Save Command Ui

* Add restriction for review command

* Change url of help window to EatMe UserGuide

* Update user guide

* Update user guide

* Increase depth of content page

* Edit SaveTodoCommand section

* Update SaveTodo section in User guide

ROHITREDDYBALAM committed:
Edit review command (#172)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* Add editreview command and associated parser

* Retain reviews for edited eateries

* Make changes to model

* Set active eatery and fix edit review command

* Add delete review command

* fix checkstyle errors

* Fix checkstyle errors

* fix edit review bug

* add tests for review class

* add Review command tests

* add review command parser tests and fix checkstyle errors

* improve code quality

* Images for dg

* Use logger

wpuffs committed:
Update user guide (add, edit, reopen, close, import, etc) (#177)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update UserGuide

* Remove trailing whitespaces

* Update user guide with tags

* Update UserGuide.adoc

* Update UserGuide.adoc

JunHongT committed:
Update developer guide with SaveTodo Command (#171)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Developer Guide with SaveTodo Command

* Update diagram for mode and save command

* Update Developer Guide with Mode And SaveTodo

* Fix merge conflict
'

JunHongT committed:
Update UI for Todo (#97)

* Implement AddTodoCommandParser for todo list

* Refractor AddressBookParser

* Modify command message for AddTodoCommandParser

* Fix checkstyle

* Create AddTodoCommandParserTest

* Fix checkstyle

* Implement SaveTodoCommand

* Implement test cases for SaveTodoCommand and SaveTodoCommandParser

* Fix checkstyle

* Fix checkstyle

* Update Ui for todolist

* Create TodoListCard.fxml and TodoListPanel.fxml

* Fix checkstyle

* Fix checkstyle

* Update eaterycard for todo mode

* Fix category label for eaterycard

* Edit EateryListCard.fxml

dvrylc committed:
Update DG with news feed feature (#174)

dvrylc committed:
Update UG for news feed and `show` command (#176)

dvrylc committed:
Increase default window size (#178)

wpuffs committed:
Update developer guide (#170)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add activity and sequence diagram for Add command

* Change sequence diagram

* Change activity diagram

* Resize activity diagram

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update photos

* Update DeveloperGuide.adoc

Update add command

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

Add stats feature and modify add feature

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update the diagrams

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update general diagrams

* Update DeveloperGuide.adoc

* Edit Storage class diagram

* Update diagrams to follow theme

* Update DeveloperGuide.adoc

Update picture links

* Add newline at EOF

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update Developer guide

* Update developer guide

* Remove "the"

JunHongT committed:
Fix bug for addtag and removetag in Todo Mode (#181)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Developer Guide with SaveTodo Command

* Update diagram for mode and save command

* Update Developer Guide with Mode And SaveTodo

* Fix merge conflict
'

* Update addtag and removetag for todo

wpuffs committed:
Fix stats bug (#184)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Implement stats command i.e. the statistics feature

* Fix line ending

* Change the logic of how stats command works

* Remove existing stats exceptions and add new ones

* Add stylesheets related to the charts and graphs generated by stats command

* Modify and add new custom charts and graphs

* Fix coding style violations

* Minor changes to exception handling

* Create tests for stats command

* Fix coding style violations

* Create tests for parsing stats command

* Fixed unhandled exception when review is less than 3

* Fix coding style violations

* Rename stats exception classes to follow convention

* Make appropriate changes based on code review

* Attempt to fix wrong line ending

* Fix CRLF line endings

* Optimize compareTo for priority queues

* THIRD attempt at fixing line ending issue

* Edit stats section of user guide to make more sense

* Fix the stats issue where most shows least data and least shows most data

* Fix label

* Remove duplicate variables

ROHITREDDYBALAM committed:
Add features to DG (#179)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* make changes to dg

* add features to dg

* fix doc errors

* add review activity diagram

* add Editreview activity diagram

* add review sequence diagram

* add sequence diagrams to dg and allow reviews to closed eateries

* Change address book to eatery list

* add new lines at eof

wpuffs committed:
Update developer guide  (#185)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Add activity and sequence diagram for Add command

* Change sequence diagram

* Change activity diagram

* Resize activity diagram

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update photos

* Update DeveloperGuide.adoc

Update add command

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

Add stats feature and modify add feature

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update the diagrams

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update general diagrams

* Update DeveloperGuide.adoc

* Edit Storage class diagram

* Update diagrams to follow theme

* Update DeveloperGuide.adoc

Update picture links

* Add newline at EOF

* Update DeveloperGuide.adoc

* Update DeveloperGuide.adoc

* Update Developer guide

* Update developer guide

* Remove "the"

* Update developer guide

dvrylc committed:
Update command result messages (#190)

* Update command result messages

* Fix test cases for new command result messages

jakq committed:
Add Reopen and Load feature in the Developer Guide (#187)

* Add Reopen and Load feature in the Developer Guide

* Remove unused test file

ROHITREDDYBALAM committed:
Dg changes (#191)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* make changes to dg

* add features to dg

* fix doc errors

* add review activity diagram

* add Editreview activity diagram

* add review sequence diagram

* add sequence diagrams to dg and allow reviews to closed eateries

* Change address book to eatery list

* add new lines at eof

* change ug pictures

* add review use case diagram

* add line at eof

dvrylc committed:
Refactor addressbook package to eatme (#186)

* Refactor addressbook package to eatme

* Improve refactor

* Fix invalid test filename

* Add more missing stuff

JunHongT committed:
Update Uml diagram for DG (#194)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Developer Guide with SaveTodo Command

* Update diagram for mode and save command

* Update Developer Guide with Mode And SaveTodo

* Fix merge conflict
'

* Update uml diagram for Mode and SaveTodo Command

JunHongT committed:
Edit user guide (#182)

* Fix bug for Save Command Ui

* Add restriction for review command

* Change url of help window to EatMe UserGuide

* Update user guide

* Update user guide

* Increase depth of content page

* Edit SaveTodoCommand section

* Update SaveTodo section in User guide

* Update UG

* Update UserGuide.adoc

- Change import command to load command
- Add short description of main and todo mode in "Introduction" section
-  Edit type in "Features" section

* Update UserGuide.adoc

- Attempt to make user guide more logical by reorganising commands
- Attempt to make user guide more consistent

* Remove trailing whitespaces

* Update UserGuide.adoc

* Update UserGuide.adoc

JunHongT committed:
create PPP [Jun Hong] (#183)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Project Portfolio

* Add images for PPP

* Update PPP

* Fix checkstyle

* Resolving merge conflict

* Resolve merge conflict

* update sequence diagram for savetodo

* Update PPP

* Remove debug statement

ROHITREDDYBALAM committed:
Fix Review parser bug (#200)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* fix review parser bug

dvrylc committed:
Update image assets (#197)

* Update UI image

* Update app icon

dvrylc committed:
Update command result to show added/edited eatery (#201)

* Update command result to show added/edited eatery

* Fix tests for new command result

JunHongT committed:
Edit save command (#189)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Project Portfolio

* Add images for PPP

* Update PPP

* Remove debuggin line

* Resolve merge conflict

* Edit wantToSave value

* Update DeveloperGuide.adoc

ROHITREDDYBALAM committed:
New ppp (#199)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* create pp and change edit review seq diagram

* Update UG

* change ppp file name

* Update ug

* Update ug

* Delete pdf

jakq committed:
Add Kai Qi's PPP (#198)

* Add Kai Qi's PPP

* Update Kai Qi's PPP

* Update Kai Qi's PPP

dvrylc committed:
Add PPP (#180)

JunHongT committed:
JunHongT PPP (#203)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Project Portfolio

* Add images for PPP

* Update PPP

* Fix checkstyle

* Resolving merge conflict

* Resolve merge conflict

* update sequence diagram for savetodo

* Update PPP

* Remove debug statement

* Rename PPP name

dvrylc committed:
Remove sample feed (#205)

ROHITREDDYBALAM committed:
Edit review bug (#206)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* fix edit review bug

JunHongT committed:
resize image for PPP (#207)

* Edit Developer Guide for Mode Command

* Update developer guide

* Update Developer Guide with SaveTodo Command

* Update uml diagram for SaveTodo Command

* Update Project Portfolio

* Add images for PPP

* Update PPP

* Fix checkstyle

* Resolving merge conflict

* Resolve merge conflict

* update sequence diagram for savetodo

* Update PPP

* Remove debug statement

* Rename PPP name

* rename PPP

* resize image

wpuffs committed:
Reorganise developer guide and add my PPP (#193)

* Remove all instances of address

* Add new command: Remark

* Update UserGuide.adoc

* Update UserGuide.adoc

* Update Ui.png

* Update UserGuide.adoc

* Update Ui.png

* Revert "Update Ui.png"

This reverts commit 09529ffa6d15f40b21807f2d1af2b6fd64b99378.

* Revert "Update UserGuide.adoc"

This reverts commit 1011cb16e1d28a6e0b104c2ab2558f326e4deca1.

* Revert "Update Ui.png"

This reverts commit 43aa00b780ab5e7ec30d320bddd7fbb157c98808.

* Revert "Update UserGuide.adoc"

This reverts commit c907562aa13235e5c18bbad56c32118429e03c55.

* Revert "Update UserGuide.adoc"

This reverts commit ea7590c2db054fb1359da7ec92c1364efa3b6da6.

* Revert "Merge branch 'tutorial-adding-command'"

This reverts commit 3f671ae4b369ecc1941d61313756e879f0c52515, reversing
changes made to 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Revert "Remove all instances of address"

This reverts commit 5e1abfd36d07a4f508e11d4db75ed375d1c7d091.

* Update UserGuide.adoc

* Update UserGuide.adoc

* Implement stats command i.e. the statistics feature

* Fix line ending

* Change the logic of how stats command works

* Remove existing stats exceptions and add new ones

* Add stylesheets related to the charts and graphs generated by stats command

* Modify and add new custom charts and graphs

* Fix coding style violations

* Minor changes to exception handling

* Create tests for stats command

* Fix coding style violations

* Create tests for parsing stats command

* Add my portfolio page

* Edit PPP

* Edit developer guide

* Reorganise commands in developer guide

* Add diagrams

* Remove existing files

* Fix merge conflicts

* Resize images

* Fix images (again)

* Minor edit

* Modify model class diagram

* Update diagram pictures

* Resize diagram and rename ppp document

* Change line ending

* Made changes to PPP

* Update developer guide and my ppp

dvrylc committed:
Fix review command, improve command result and fix statistics

dvrylc committed:
Standardize messages and fix review sort

dvrylc committed:
Implement error detection for feed addresses

ROHITREDDYBALAM committed:
Ugchanges (#208)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* change ug

ROHITREDDYBALAM committed:
Merge pull request #209 from dvrylc/fix-final-bugs

Fix final bugs

ROHITREDDYBALAM committed:
Ugchanges (#210)

* remove email field

* Remove email field

* Remove phone field

* Remove phone attribute

* Remove phone, email from test  case

* Resolve merge conflicts

* Add API key

* change ug

* change review command message

* fix checkstyle errors

* Update review message

dvrylc committed:
Update docs

JunHongT committed:
Merge pull request #211 from dvrylc/update-docs

Update docs
